tests/trainer/test_optimizers.py

```diff
@@ -126,55 +126,60 @@ def test_multi_optimizer_with_scheduling_stepping(tmpdir):
         'lr for optimizer 2 not adjusted correctly'
 
 
-def test_reduce_lr_on_plateau_scheduling_missing_monitor(tmpdir):
-
-    hparams = EvalModelTemplate.get_default_hparams()
-    model = EvalModelTemplate(**hparams)
-    model.configure_optimizers = model.configure_optimizers__reduce_lr_on_plateau
-
-    # fit model
-    trainer = Trainer(
-        default_root_dir=tmpdir,
-        max_epochs=1,
-        limit_val_batches=0.1,
-        limit_train_batches=0.2,
-    )
-
-    m = '.*ReduceLROnPlateau requires returning a dict from configure_optimizers.*'
-    with pytest.raises(MisconfigurationException, match=m):
+def test_reducelronplateau_with_no_monitor_raises(tmpdir):
+    """
+    Test exception when a ReduceLROnPlateau is used with no monitor
+    """
+    model = EvalModelTemplate()
+    optimizer = torch.optim.Adam(model.parameters())
+    model.configure_optimizers = lambda: ([optimizer], [torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer)])
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True)
+    with pytest.raises(
+        MisconfigurationException, match='`configure_optimizers` must include a monitor when a `ReduceLROnPlateau`'
+    ):
         trainer.fit(model)
 
 
-def test_reduce_lr_on_plateau_scheduling(tmpdir):
-    hparams = EvalModelTemplate.get_default_hparams()
-
-    class TestModel(EvalModelTemplate):
-
-        def configure_optimizers(self):
-            optimizer = torch.optim.Adam(self.parameters(), lr=self.learning_rate)
-            lr_scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer)
-            return {'optimizer': optimizer, 'lr_scheduler': lr_scheduler, 'monitor': 'early_stop_on'}
+def test_reducelronplateau_with_no_monitor_in_lr_scheduler_dict_raises(tmpdir):
+    """
+    Test exception when lr_scheduler dict has a ReduceLROnPlateau with no monitor
+    """
+    model = EvalModelTemplate()
+    optimizer = torch.optim.Adam(model.parameters())
+    model.configure_optimizers = lambda: {
+        'optimizer': optimizer,
+        'lr_scheduler': {
+            'scheduler': torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer),
+        },
+    }
```

Comment:
add this to `ComfigureOptimizersPool`?

Reply:
I don't see a point in adding it to the pool if it is not reused in other tests. It just adds extra boilerplate. This way, if the test is updated/removed, you only have to change code in one single place.

Reply:
hmm. ok :+1: